### PR TITLE
ISPN-14545 SIFS Compactor does not properly shut down but the index t…

### PR DESCRIPTION
**Pull request by wburns**
…hinks it is okay

https://issues.redhat.com/browse/ISPN-14545

### CODE REVIEW

**Comment by tristantarrant** [comment]
A few SIFS failures

File: core/src/main/java/org/infinispan/persistence/sifs/Compactor.java [hunk 1]
```diff
@@ -272,10 +299,18 @@ private void resumeAfterPause() {
...
       processor.onComplete();
-      Util.close(logFile);
-      logFile = null;
+      // The stopped CompletableFuture is completed in onComplete or onError callback for the processor, so this will
+      // return after all compaction calls are completed
+      stopped.join();
+      if (logFile != null) {
+         Util.close(logFile);
+         // Complete the file, this file should not be compacted
+         completeFile(logFile.fileId, currentOffset, nextExpirationTime);
```
Comment (jabolina) [inline core/src/main/java/org/infinispan/persistence/sifs/Compactor.java:311]: Nitpicking here as it doesn't actually affect the execution, but a `false` at the end to not even invoke `schedule`?

**Review by jabolina — COMMENTED** [review]
Just small things.

File: core/src/main/java/org/infinispan/persistence/sifs/Compactor.java (hunk 1 above)
Comment (wburns) [reply core/src/main/java/org/infinispan/persistence/sifs/Compactor.java:311]: Sure, it would be more explicit then.

**Comment by wburns** [comment]
Updated to address comments

**Review by jabolina — APPROVED** [review]

**Comment by jabolina** [comment]
Merged, thanks @wburns

**Comment by tristantarrant** [comment]
> Should SoftIndexFileStoreTestUtils.java be in main rather than test?

As it's a test class, it is correctly in `test`. We publish test jars to Maven with the `test-jar` classifier in case you want to use it.

**Comment by FireBurn** [comment]
Sorry it was just because dataDirectorySize was used by the fix in ISPN-15184 `testExpirationCompactionOnLogFile` is just testing the fix though isn't it?

**Pull request merged**